### ISSUE TITLE
Switch should be announced as Switch

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#14378]

**Comment by jonahwilliams** [comment]
I did some research on this that leads me to believe that there are definitely some semantic differences between a switch and checkbox, despite the fact that all switches are checkboxes.

- A switch implies an immediate effect; turning off wifi versus a form option which requires a secondary action to take effect.
- A switch implies a mutually exclusive state for a single option, a checkbox can be part of a group of options.
- A switch doesn't support indeterminate state.

For reference:

[UWP Guidelines](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/toggles)
"Use toggle switch controls to present users with two mutually exclusive options (such as on/off), where choosing an option provides immediate results."

[MDN aria-switch](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_switch_role)
"The ARIA switch role is functionally identical to the checkbox role" 
"switch does not support the value mixed"

**Comment by goderbauer** [comment]
Thanks for the research! Did it say anything specific about how they should be announced in a11y mode?

**Comment by jonahwilliams** [comment]
on - off seems the be the common theme, but the aria-switch documentation states that some screen readers may ignore that and treat it as a checkbox

**Comment by jonahwilliams** [comment]
After investigating this on iOS I've found there is no trait we can use to have our semantics objects implement a switch.  Switch itself implements a private protocol shared among controls, so It looks like we have few options here.

**Comment by jonahwilliams** [comment]
Maybe we could build our own semantics object which implements enough behaviors to act like a switch.  This would require doing things like adding localizations for switch/on/off to the engine (unless we could get the value from iOS somehow)

**Comment by jonahwilliams** [comment]
This is done on Android, still investigating iOS implementation

**Comment by clocksmith** [comment]
blocked by https://github.com/flutter/flutter/issues/19481

**Comment by TahaTesser** [comment]
On testing on iOS with voice over enabled, it's announced as Switch button -> "Switch. button On, double tap to toggle settings" when  using `CupertinoSwitch` and `Switch`


<details> 
<summary>Code Sample</summary> 

```
import 'package:flutter/cupertino.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Material App',
      theme: ThemeData.dark(),
      home: Home(),
    );
  }
}

class Home extends StatefulWidget {
  @override
  _HomeState createState() => _HomeState();
}

class _HomeState extends State<Home> {
  bool _value = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Material App Bar'),
      ),
      body: Center(
        child: CupertinoSwitch(
          value: _value,
          onChanged: (bool newValue) {
            setState(() {
              _value = newValue;
            });
          },
        ),
      ),
      floatingActionButton:
          FloatingActionButton(child: Icon(Icons.add), onPressed: () {}),
    );
  }
}

``` 

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel dev, 1.21.0-1.0.pre, on Mac OS X 10.15.6 19G73, locale
    en-GB)
    • Flutter version 1.21.0-1.0.pre at /Users/taha/Code/flutter_dev
    • Framework revision f25bd9c55c (8 days ago), 2020-07-14 20:26:01 -0400
    • Engine revision 99c2b3a245
    • Dart version 2.9.0 (build 2.9.0-21.0.dev 20bf2fcf56)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/taha/Code/sdk
    • Platform android-29, build-tools 29.0.3
    • ANDROID_HOME = /Users/taha/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.6)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.6, Build version 11E708
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 47.1.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.47.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.12.2

[✓] Connected device (4 available)
    • SM M305F (mobile) • 32003c30dc19668f • android-arm64  • Android 10 (API
      29)
    • macOS (desktop)   • macos            • darwin-x64     • Mac OS X 10.15.6
      19G73
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      84.0.4147.89

• No issues found!
```

</details> 

Related https://github.com/flutter/flutter/issues/16395

**Comment by goderbauer** [comment]
The current behavior is the expected one. Closing.
